src/components/data/DataTable.js

```diff
@@ -316,7 +318,7 @@ const DataTable = props => {
               initialX,
               initialY,
               sort,
-              numFixedColumns: 2,
+              numFixedColumns: visibleColumns.length > 2 ? 2 : 0,
```

Comment:
```suggestion
              numFixedColumns: visibleColumns.length > 0 ? 2 : 0,
```

`visibleColumns` is based on `columnSettings` and does not include the first two columns (checkbox and row ID). As is, this would only fix the first two columns when there are more than 4 columns in the table.